docs/csharp/whats-new/csharp-10.md

```diff
@@ -168,3 +169,58 @@ You can learn more about this feature in the article on [Caller information attr
 ## Enhanced #line pragma
 
 C# 10 supports a new format for the `#line` pragma. You likely won't use the new format, but you'll see its effects. The enhancements enable more fine-grained output in domain-specific languages (DSLs) like Razor. The Razor engine uses these enhancements to improve the debugging experience. You'll find debuggers can highlight your Razor source more accurately. To learn more about the new syntax, see the article on [Preprocessor directives](../language-reference/preprocessor-directives.md#error-and-warning-information) in the language reference. You can also read the [feature specification](~/_csharplang/proposals/csharp-10.0/enhanced-line-directives.md#examples) for Razor based examples.
+
+## Generic attributes
+
+> [!IMPORTANT]
+> *Generic attributes* is a preview feature. You must [set the `<LangVersion>` to `Preview`](../language-reference/compiler-options/language.md#langversion) to enable this feature. This feature may change before its final release.
+
+You can declare a [generic class](../programming-guide/generics/generic-classes.md) whose base class is <xref:System.Attribute?displayProperty=fullName>. This provides a more convenient syntax for atributes that require a <xref:System.Type?displayProperty=nameWithType> parameter. Previously, you'd need to write an attribute that takes a `Type` as its parameter:
+
+```csharp
+public class TypeAttribute : Attribute
+{
+   public TypeAttribute(Type t) => ParamType = t;
+
+   public Type ParamType { get; }
+}
+```
+
+To apply the attribute, you use the [`typeof`](../language-reference/operators/type-testing-and-cast.md#typeof-operator) operator:
+
+```csharp
+[TypeAttribute(typeof(string))] 
+public string Method() => default;
+```
+
+Using this feature, you can use a generic attribute instead:
+
+```csharp
+public class GenericAttribute<T> : Attribute { }
+```
+
+Then, specify the type parameter to use the attribute:
+
+```csharp
+[GenericAttribute<string>()]
+public string Method() => default;
+```
+
+You can apply a fully closed constructed generic attribute. In other words, all type parameters must be specified. For example, the following is not allowed:
+
+```csharp
+public class GenericType<T>
+{
+   [GenericAttribute<T>()] // Not allowed! generic attributes must be fully closed types.
+   public string Method() => default;
+}
+```
+
+The type arguments must satisfy the same restrictions as the [`typeof`](../language-reference/operators/type-testing-and-cast.md#typeof-operator) operator. Types that require metadata annotations aren't allowed. Examples include the following:
+
+- `dynanmic`
+- `nint`, `nuint`
```

Comment:
@RikkiGibson These two work with `typeof`. Should they work here?

Reply:
What happened here is: we took generic attributes as an opportunity to "tighten up" the rules on using type arguments that can't actually be represented in metadata. However, for compat, it proved to be too difficult at least for .NET 6 to bring the same rules back over to typeof, even though typeof has similar limitations as generic attribute type arguments.

Related discussion: dotnet/roslyn#54804

Reply:
Thanks @RikkiGibson That discussion helped. I added some clarifications and what types to use instead.